multiqc/modules/ivar/ivar.py

```diff
@@ -0,0 +1,167 @@
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import bargraph, heatmap
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name = 'iVar',
+            anchor = 'iVar',
+            href = 'https://github.com/andersen-lab/ivar',
+            info = "is a computational package that contains functions broadly useful for viral amplicon-based sequencing."
+        )
+
+        # Find and load iVar trim results
+        self.ivar_data = dict()
+        self.parsed_primers = dict()
+        for f in self.find_log_files('ivar/trim', filehandles=True):
+            parsed_data = self.parse_ivar(f)
+            if parsed_data is not None and len(parsed_data) > 0:
+                self.ivar_data[f['s_name']] = parsed_data
+                self.add_data_source(f, f['s_name'])
+        for f in self.find_log_files('ivar/trim'):
+            parsed_primers = self.parse_ivar_primer_stats(f)
+            if parsed_primers is not None and len(parsed_primers) > 0:
+                self.parsed_primers[f['s_name']] = parsed_primers
+                self.add_data_source(f, f['s_name'])
+
+        # Filter to strip out ignored sample names
+        self.ivar_data = self.ignore_samples(self.ivar_data)
+
+        # Warning when no files are found
+        if len(self.ivar_data) == 0:
+            raise UserWarning
+
+        # Write parsed data to a file
+        self.write_data_file(self.ivar_data, 'multiqc_ivar_summary')
+        
+        #Primers too
+        self.write_data_file(self.parsed_primers, 'multiqc_ivar_primers')
+        
+        #Found reports or not?
+        log.info("Found {} reports".format(len(self.ivar_data)))
+
+        # Basic Stats Table
+        self.ivar_general_stats_table()
+
+        #Heatmap info
+        self.status_heatmap()
+        
+    # Parse an ivar report
+    def parse_ivar(self, f):
+        parsed_data = dict()
+        regexes = {
+            'mapped_reads': r'(?:Found\s)(\d+)(?:\smapped)',
+            'total_reads': r'(?:Trimmed primers from )(?:\d+\.\d+\% \()?(\d+)',
+            'reads_outside_primer_region': r'^(?:\d+\.\d+\% \()?(\d+)(?:\))?(?:.*[of]?)reads\s(?:that\s)?started',
+            'reads_too_short_after_trimming': r'^(?:\d+\.\d+\% \()?(\d+)(?:\))?(?:.*[of]?)reads\swere(?: quality trimmed | shortened)'
+        }
+        for l in f['f']:
+            # Search regexes for stats
+            for k, r in regexes.items():
+                match = re.search(r, l)
+                if match:
+                    parsed_data[k] = int(match.group(1))
+        
+        return parsed_data
+
+    #Parse Primer stats appropriately
+    def parse_ivar_primer_stats(self, f):
```

Comment:
Any reason that this has to be a separate parsing function? This is almost identical to the above, so might be cleaner to just loop through the file once.

Reply:
I'll take a look at it when I come to merge 👍